src/net_processing.cpp

```diff
@@ -3388,6 +3411,17 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
             // they may wish to request compact blocks from us
             m_connman.PushMessage(&pfrom, msgMaker.Make(NetMsgType::SENDCMPCT, /*high_bandwidth=*/false, /*version=*/CMPCTBLOCKS_VERSION));
         }
+
+        if (m_txreconciliation) {
+            if (!peer->m_wtxid_relay || !m_txreconciliation->IsPeerRegistered(pfrom.GetId())) {
+                // We could have optimistically pre-registered/registered the peer.
+                // Forget about the reconciliation state if peer hasn't announced WTXIDRELAY
+                // or sent us SENDTXRCNCL.
```

Comment:
I think we `RegisterPeer`a peer on their SENDTXNRCNCL message reception, comment could be updated to drop out the pre-registration reference and make it clear both conditions must be checked: announcement of WTXIDRELAY and announcement of SENDTXRCNCL?

Reply:
I see the comment is not ideal, but i want to understand you concern fully before proceeding with the change.

To help me better understand your concern, could you provide a minimal diff in the comment text you'd suggest?

Reply:
Something like:

"We could have optimistically pre-registered the peer. Forget about the reconciliation state if peer hasn't announced WTXIDRELAY and sent us SENDTXRCN"

Comment matching exactly the code from my understanding.